### PR TITLE
Feature/PTWO-213 typescript lit element migration

### PR DESCRIPTION
**Pull request by pgoc**
After merging tasks to do:

- [ ] npm version major
- [ ] git push origin --tags
- [ ] git push origin master
- [ ] npm publish

### CODE REVIEW

**Review by marksmits — APPROVED** [review]

File: exmg-markdown-editor.ts [hunk 1]
```diff
@@ -0,0 +1,927 @@
...
+*  `--exmg-markdown-editor-toolbar-seperator-color` | Toolbar seperator color | `#ddd`
+*  `--exmg-markdown-editor-code` | CodeMirror root mixin | `{}`
+*  `--exmg-markdown-editor-code-hover` | Editor code part hover background color | `white`
+*  `--exmg-markdown-editor-code-focused` | CodeMirror editor focused mixin | `{}`
+*
+*  # Events:
+*  - value-change - where detail is current markdown value
+*  - exmg-markdown-editor-fullscreen where detail is boolean with current fullscreen state
+*
+* @customElement
```
Comment (markoffk) [inline exmg-markdown-editor.ts:117]: I think these annotations are for polymer element, not for lit element

File: exmg-markdown-editor.ts [hunk 2]
```diff
@@ -0,0 +1,927 @@
...
+  private normalizedToolBarConfig: Map<ToolBarOption, ToolBarConfigItem> = new Map();
+
+  private dispatchMarkdownUpdatedDebounce: (cb?: Function) => void = debounce(300);
+
+  private isElementInitialized: boolean = false;
+
+  /**
+   * When ready check if markdown property is set or otherwise look for script tag
+   */
+  private ready() {
```
Comment (markoffk) [inline exmg-markdown-editor.ts]: return void

File: exmg-markdown-editor.ts [hunk 3]
```diff
@@ -0,0 +1,927 @@
...
+    });
+
+    afterNextRender(this, () => this.updateDocHistory());
+
+    this.codeMirrorEditor = codeMirrorEditor;
+
+    return codeMirrorEditor;
+  }
+
+  private replaceRangeLine(text: string, lineNumber: number) {
```
Comment (markoffk) [inline exmg-markdown-editor.ts]: return void

File: exmg-markdown-editor.ts [hunk 4]
```diff
@@ -0,0 +1,927 @@
...
+    this.codeMirrorEditor!.getDoc().replaceRange(text, {line: lineNumber, ch: 0}, {line: lineNumber, ch: 99999999999999});
+  }
+
+  private insertAtCursor(text: string, selectionOffset?: number, selectionLength?: number) {
```
Comment (markoffk) [inline exmg-markdown-editor.ts]: return void

File: exmg-markdown-editor.ts [hunk 5]
```diff
@@ -0,0 +1,927 @@
...
+
+  private toggleBlockquote(event?: Event): void {
+    if (event) {
+      event.preventDefault();
+    }
+
+    this.processLine('quote', '>');
+  }
+
+  private toggleUnorderedList(event? :Event): void {
```
Comment (markoffk) [inline exmg-markdown-editor.ts]: please move ':' closer to param name

File: exmg-markdown-editor.ts [hunk 6]
```diff
@@ -0,0 +1,927 @@
...
+    }
+
+    if (this.isSelectionInline()) {
+      this.processBlock('inline-code');
+    } else {
+      this.processBlock('code', true);
+    }
+  }
+
+  private undo(event? :Event): void {
```
Comment (markoffk) [inline exmg-markdown-editor.ts]: please move ':' closer to param name

File: exmg-markdown-editor.ts (hunk 2 above)
Comment (pgoc) [reply exmg-markdown-editor.ts]: done

File: exmg-markdown-editor.ts (hunk 4 above)
Comment (pgoc) [reply exmg-markdown-editor.ts]: done

File: exmg-markdown-editor.ts (hunk 5 above)
Comment (pgoc) [reply exmg-markdown-editor.ts]: done

File: exmg-markdown-editor.ts (hunk 6 above)
Comment (pgoc) [reply exmg-markdown-editor.ts]: done

File: exmg-markdown-editor.ts (hunk 1 above)
Comment (pgoc) [reply exmg-markdown-editor.ts:117]: More of them are just JsDoc so can be used regardless of framework.
I will keep it as it was

File: exmg-markdown-editor.ts (hunk 3 above)
Comment (pgoc) [reply exmg-markdown-editor.ts]: done

**Review by markoffk — APPROVED** [review]

**Pull request merged**